teps/0096-pipelines-v1-api.md

```diff
@@ -174,11 +182,110 @@ V1 work should prioritize stabilizing existing beta features that support the [u
+- Implement [TEP-0073: Simplify metrics](https://github.com/tektoncd/community/blob/main/teps/0073-simplify-metrics.md).
+  - [TEP-0006: Tekton metrics](./0006-tekton-metrics.md) is out of scope because it discusses adding more granular metrics. More granular metrics may be added later.
+- Ensure metrics are accurate and mark all metrics as "alpha" or "stable". 
+
+### Stability
+
+- Track bug fixes and small backwards incompatible changes elsewhere (in a spreadsheet, milestone, or similar).
+- Implement conversion webhook for v1beta1 objects.
+
+#### CRD Stability Levels
+
+- The Task, TaskRun, Pipeline, and PipelineRun CRDs should be stabilized.
+- Because Custom Task is mission critical to our extensibility use case, Run should be upgraded to beta stability.
+  - See [Custom Tasks Beta](https://github.com/tektoncd/pipeline/issues/4313) for more information on work required to upgrade Custom Tasks to beta.
+- Keep the ClusterTask CRD at beta stability.
+  - ClusterTask functionality will likely be replaced by [remote resolution](./0060-remote-resource-resolution.md), but there's no plan
+  to replace this functionality at a beta level of stability before releasing a V1 API.
+  - We will decide whether to stabilize or remove ClusterTask pending community feedback on ClusterTask and remote resolution.
+
+| CRD         | Current level | Proposed level |
+| ----------- | ------------- | -------------- |
+| Task        | beta          | stable         |
+| TaskRun     | beta          | stable         |
+| Pipeline    | beta          | stable         |
+| PipelineRun | beta          | stable         |
+| ClusterTask | beta          | beta           |
+| Run         | alpha         | beta           |
+
+#### API Changes
+
+- Implement [TEP-0094: Improve Task Resources UX](https://github.com/tektoncd/community/pull/588).
+- Fix [pain points](https://github.com/tektoncd/pipeline/issues/3792) associated with TaskRun and PipelineRun Status.
+- Consider replacing any Task or Pipeline fields that should support parameterization with Strings and performing our own validation.
+  - See [Handling parameter interpolation in fields not designed for it](https://github.com/tektoncd/pipeline/issues/1530) for more details.
+
+#### Deprecations
+
+- Deprecate `PipelineRun.Timeout` and upgrade `PipelineRun.Timeouts` to beta.
+- Deprecate `Pipeline.Tasks[].Conditions`. Conditions have been replaced by when expressions, which are already in beta.
+- Because PipelineResources are deprecated, remove the `Resources` fields of `Task`, `TaskRun`, `Pipeline`, and `PipelineRun`.
+    - We will proceed with [TEP-0074: Deprecate PipelineResources](https://github.com/tektoncd/community/blob/main/teps/0074-deprecate-pipelineresources.md)
+    independently of Pipelines V1, and merging this TEP as implementable isn't a V1 blocker.
+    However, we should still provide PipelineResource users with a migration path to the V1 API.
```

Comment:
I am not sure I understand this.

Reply:
This was also discussed at yesterday's API WG. Right now, we aren't sure how many users are using PipelineResources, and what kind of replacement would allow them a migration path to v1. We decided to avoid descoping TEP-0074, as the requirements to mark this TEP as implementable should still be part of the deprecation of PipelineResources, but the requirements are a large chunk of work to accomplish before releasing a v1 API. I will likely send out a survey to try to understand better what a "migration path" would look like, and can update the TEP to reflect that this hasn't yet been decided.

Reply:
👌🏼

Reply:
Updated to note that this is still under discussion.